src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py

```diff
@@ -148,7 +148,11 @@ def builtin_if(
     expr = "({1} if {0} else {2})".format(*internals)
     node_type = transformer.node_types[id(node)]
     assert isinstance(node_type, itir_typing.Val)
-    type_ = itir_type_as_dace_type(node_type.dtype)
+    try:
+        type_ = itir_type_as_dace_type(node_type.dtype)
+    except NotImplementedError:
+        # the if expression is formatted with the return value as first argument
+        type_ = args[1].dtype
```

Comment:
Is this just a workaround for a bug in type inference?

Reply:
Same here.